index.ts

```diff
@@ -68,9 +68,97 @@ async function toMatchSvgSnapshot(
   }
 }
 
+async function toMatchMultipleSvgSnapshot(
+  // biome-ignore lint/suspicious/noExplicitAny: bun doesn't expose
+  this: any,
+  receivedMaybePromise: string[] | Promise<string[]>,
+  testPathOriginal: string,
+  svgNames: string[],
+): Promise<MatcherResult> {
+  const passed = []
+  const failed = []
+  for (let index = 0; index < svgNames.length; index++) {
+    const svgName = svgNames[index]
+    const received = await receivedMaybePromise
+    const testPath = testPathOriginal.replace(/\.test\.tsx?$/, "")
+    const snapshotDir = path.join(path.dirname(testPath), "__snapshots__")
+    const snapshotName = svgName
+      ? `${svgName}.snap.svg`
+      : `${path.basename(testPath)}.snap.svg`
+    const filePath = path.join(snapshotDir, snapshotName)
+
+    if (!fs.existsSync(snapshotDir)) {
+      fs.mkdirSync(snapshotDir, { recursive: true })
+    }
+
+    const updateSnapshot =
+      process.argv.includes("--update-snapshots") ||
+      process.argv.includes("-u") ||
+      Boolean(process.env.BUN_UPDATE_SNAPSHOTS)
+
+    if (!fs.existsSync(filePath) || updateSnapshot) {
+      console.log("Writing snapshot to", filePath)
+      fs.writeFileSync(filePath, received[index])
+      passed.push({
+        message: `Snapshot ${svgName} created at ${filePath}`,
+        pass: true,
+      })
+      continue
+    }
+
+    const existingSnapshot = fs.readFileSync(filePath, "utf-8")
+
+    const result = await looksSame(
+      Buffer.from(received[index]),
+      Buffer.from(existingSnapshot),
+      {
+        strict: false,
+        tolerance: 2,
+      },
+    )
+
+    if (result.equal) {
+      passed.push({
+        message: `Snapshot ${svgName} matches`,
+        pass: true,
+      })
+      continue
+    }
+
+    const diffPath = filePath.replace(".snap.svg", ".diff.png")
+    await looksSame.createDiff({
+      reference: Buffer.from(existingSnapshot),
+      current: Buffer.from(received[index]),
+      diff: diffPath,
+      highlightColor: "#ff00ff",
+    })
+
+    failed.push({
+      message: `Snapshot ${svgName} does not match. Diff saved at ${diffPath}`,
+      pass: false,
+    })
+  }
+  if (failed.length === 0) {
+    let messages = ""
```

Comment:
Just declare this once, its currently declared twice which isnt necessary

Prefer the name "aggregatedMessage", messages is plural so people would think its an array